porcupine/plugins/text_google_search.py

```diff
@@ -0,0 +1,21 @@
+"""Search selected text on Google."""
+
+import tkinter as tk
+from porcupine import menubar, tabs
+import webbrowser
+
+def google_search(tab: tabs.FileTab) -> None:
+    try:
+        selected_text_tuple = tab.textwidget.tag_ranges(tk.SEL)
+        # Check if text is selected to search on google
+        if len(selected_text_tuple)>0:
+            selected_text = tab.textwidget.get(*selected_text_tuple)
+            url = "https://www.google.com/search?q={}".format(selected_text)
+            webbrowser.open_new_tab(url)
+        else:
+            tk.messagebox.showwarning("Warning","Please select text!")
+    except:
+        tk.messagebox.showerror("Error", "Something went wrong!")
```

Comment:
What's the purpose of these message boxes? I think it would be better without them:
- If you accidentally press Ctrl+G when you didn't mean to google anything, you get a message box saying "Please select text!". How are you supposed to know that it was because Porcupine tried to do a Google search? It would be better to just not bother the user with unnecessary message boxes.
- I don't like "Something went wrong" error messages. They don't tell what went wrong. Also, the user will notice if something went wrong, even without message box, when their browser doesn't open. If you don't use `try`/`except` at all, Porcupine will print the full error message to terminal and write it to a log file, in case the user wants to figure out why Ctrl+G isn't working.

Reply:
Warnings are deleted in commit: https://github.com/Akuli/porcupine/pull/1024/commits/4ec19862ec918996f7091e86d98b887f01d5ecea